src/mlpack/core/cv/purged_k_fold_cv.hpp

```diff
@@ -0,0 +1,282 @@
+/**
+ * @file core/cv/purged_k_fold_cv.hpp
+ * @author Felix Patschkowski
+ *
+ * purged k-fold cross-validation.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_CV_PURGED_K_FOLD_CV_HPP
+#define MLPACK_CORE_CV_PURGED_K_FOLD_CV_HPP
+
+#include "k_fold_cv_base.hpp"
+
+namespace mlpack {
+
+/**
+ * A purged k-fold CV is suitable for time series with informational
+ * overlap from the training set into the validation set and vice
+ * versa. This CV will remove items from the training set that overlap
+ * with the validation set.
+ */
+template<typename MLAlgorithm,
+         typename Metric,
+         typename MatType = arma::mat,
+         typename PredictionsType =
+             typename MetaInfoExtractor<MLAlgorithm, MatType>::PredictionsType,
+         typename WeightsType =
+             typename MetaInfoExtractor<MLAlgorithm, MatType,
+                 PredictionsType>::WeightsType>
+class PurgedKFoldCV :
+    private KFoldCVBase<
+        PurgedKFoldCV<MLAlgorithm, Metric, MatType, PredictionsType, WeightsType>,
+        MLAlgorithm,
+        Metric,
+        MatType,
+        PredictionsType,
+        WeightsType>
+{
+ public:
+  /**
+    * This constructor can be used for regression algorithms and for binary
+    * classification algorithms.
+    *
+    * @param k Number of folds (should be at least 2).
+    * @param[in] embargoPercentage Percentage of the dataset size that is used
+    *                              to extend the duration of the intervals in
+    *                              the validation set to remove elements from
+    *                              the training set after the validation set
+    *                              (embargo). 
+    * @param xs Data points to cross-validate on.
+    * @param ys Predictions (labels for classification algorithms and responses
+    *     for regression algorithms) for each data point.
+    * @param[in] intervals A `2 x m` matrix, where each column stores the start
+    *                      and end sample of an interval.

```

Comment:
Just to check for my understanding: this is the same setup as for the sequential bootstrap, right?  So `xs` has `n` columns (e.g. there are `n` data points), and `intervals` should also have `n` columns (not `m`).  If that's true, we should probably update the documentation to say `2 x xs.n_cols` for the size, to be clear.

Reply:
Good spot, already changed this in the latest code